hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -333,18 +333,18 @@ public Pair<Option<String>, JavaRDD<WriteStatus>> syncOnce() throws IOException
     // Refresh Timeline
     refreshTimeline();
 
-    Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> srcRecordsWithCkpt = readFromSource(commitTimelineOpt);
+    ReadResult readResult = readFromSource(commitTimelineOpt);
```

Comment:
Integ test module build failed due to
```
Error:  /home/runner/work/hudi/hudi/hudi-integ-test/src/main/java/org/apache/hudi/integ/testsuite/HoodieDeltaStreamerWrapper.java:[81,34] incompatible types: org.apache.hudi.utilities.deltastreamer.DeltaSync.ReadResult cannot be converted to org.apache.hudi.common.util.collection.Pair<org.apache.hudi.utilities.schema.SchemaProvider,org.apache.hudi.common.util.collection.Pair<java.lang.String,org.apache.spark.api.java.JavaRDD<org.apache.hudi.common.model.HoodieRecord>>>
```
Looks like we need to change the HoodieDeltaStreamerWrapper in the test to provide same return type.